python/pyarrow/tests/test_cython.py

```diff
@@ -80,6 +80,7 @@ def check_cython_example_module(mod):
         mod.cast_scalar(scal, pa.list_(pa.int64()))
 
 
+@pytest.mark.numpy
```

Comment:
Hmm, is there anything that requires NumPy in this test?

Reply:
there is an import numpy on the cython `setup_template`, it fails with:
```
pyarrow/tests/test_cython.py::test_cython_api Traceback (most recent call last):
  File "/tmp/pytest-of-raulcd/pytest-243/test_cython_api0/setup.py", line 5, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
FAILED
```
If I try to remove `numpy` (which I don't think we want on this test) the `pyarrow_cython_example` file also fails to compile with:

```
x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/raulcd/code/arrow/python/pyarrow/include -I/home/raulcd/code/no-numpy/include -I/usr/include/python3.10 -c pyarrow_cython_example.cpp -o build/temp.linux-x86_64-3.10/pyarrow_cython_example.o -std=c++17
pyarrow_cython_example.cpp:1380:10: fatal error: numpy/halffloat.h: No such file or directory
 1380 | #include "numpy/halffloat.h"
      |          ^~~~~~~~~~~~~~~~~~~
```

Reply:
I see, so NumPy is still required for Cython integration until we make NumPy optional in our headers too (but we might not care much about this).

Reply:
Can you just add a comment about it?

Reply:
My understanding is that this PR makes numpy optional at _runtime_, but numpy is indeed still a build-time dependency. But indeed, that doesn't seem to be a big priority as at build time numpy should be a relatively easy dependency to install.

Reply:
I've added a comment